### PR TITLE
Fix upsert msgid

### PR DESCRIPTION
**Pull request by smellthemoon**
#22563 
/kind bug

### CODE REVIEW

File: internal/proxy/task_delete.go [hunk 1]
```diff
@@ -252,10 +254,17 @@ func (dt *deleteTask) Execute(ctx context.Context) (err error) {
 		ts := dt.deleteMsg.Timestamps[index]
 		_, ok := result[key]
 		if !ok {
+			msgid, err := dt.idAllocator.AllocOne()
+			if err != nil {
+				log.Info("Fail to alloc msgid when Execute in delete", zap.Error(err))
+				return err
```
Comment (jiaoew1991) [inline internal/proxy/task_delete.go:260]: error should be wrapped

File: internal/proxy/task_delete.go [hunk 2]
```diff
@@ -252,10 +254,17 @@ func (dt *deleteTask) Execute(ctx context.Context) (err error) {
...
+			}
 			sliceRequest := msgpb.DeleteRequest{
 				Base: commonpbutil.NewMsgBase(
 					commonpbutil.WithMsgType(commonpb.MsgType_Delete),
-					commonpbutil.WithMsgID(dt.deleteMsg.Base.MsgID),
+					// msgid of delete msg must be set
+					// or it will be seen as duplicated msg in mq
+					commonpbutil.WithMsgID(msgid),
```
Comment (jiaoew1991) [inline internal/proxy/task_delete.go:267]: add a UT for this case?

File: internal/util/merr/errors.go [hunk 3]
```diff
@@ -98,6 +98,9 @@ var (
 	// Metrics related
 	ErrMetricNotFound = newMilvusError("MetricNotFound", 1200, false)
 
+	// Allocator related
+	ErrIDAllocatorFailed = newMilvusError("IDAllocatorFailed", 1300, false)
```
Comment (yah01) [inline internal/util/merr/errors.go:102]: We don't need to add an error here, as failed to allocate a ID/timestamp is not a leaf error, the cause should be returned by the allocator itself

File: internal/proxy/task_upsert.go [hunk 4]
```diff
@@ -415,10 +416,18 @@ func (it *upsertTask) deleteExecute(ctx context.Context, msgPack *msgstream.MsgP
 		ts := it.upsertMsg.DeleteMsg.Timestamps[index]
 		_, ok := result[key]
 		if !ok {
+			msgid, err := it.idAllocator.AllocOne()
+			if err != nil {
+				return merr.WrapErrIDAllocatorFailed("Fail to alloc msgid when deleteExecute in upsert")
```
Comment (yah01) [inline internal/proxy/task_upsert.go:421]: The original error lost, just `errors.Wrap(err, "failed to allocate MsgID for delete of upsert")`

File: internal/proxy/task_delete.go [hunk 5]
```diff
@@ -252,10 +255,16 @@ func (dt *deleteTask) Execute(ctx context.Context) (err error) {
 		ts := dt.deleteMsg.Timestamps[index]
 		_, ok := result[key]
 		if !ok {
+			msgid, err := dt.idAllocator.AllocOne()
+			if err != nil {
+				return merr.WrapErrIDAllocatorFailed("Fail to alloc msgid when Execute in delete")
```
Comment (yah01) [inline internal/proxy/task_delete.go:260]: The original error lost, just `errors.Wrap(err, "failed to allocate MsgID for delete")`

**Review by yah01 — CHANGES_REQUESTED** [review]

**Review by jiaoew1991 — APPROVED** [review]
/lgtm

**Pull request merged**